addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/CrossDomainScriptInclusionScanRuleUnitTest.java

```diff
@@ -362,4 +363,58 @@ void crossDomainScriptInContextLow() throws HttpMalformedHeaderException {
                 alertsRaised.get(0).getEvidence(),
                 equalTo("<script src=\"https://www.otherDomain.com/script2\"/>"));
     }
+
+    @Test
+    void dontRaiseIssueWhenLinkIsTrustedDomain() throws HttpMalformedHeaderException {
+        // Given
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET https://www.example2.com/test/ HTTP/1.1");
+        rule.getConfig()
+                .setProperty(RuleConfigParam.RULE_DOMAINS_TRUSTED, "https://www.example2.com/.*");
```

Comment:
We should only need to trust 3rd-party domains.

Reply:
What should I change?

Reply:
It should be different than the host being accessed and should match the script element. Since they are the same already change the host of the message to e.g. `www.example.com`.

Reply:
Done